packages/plugin-scripts/src/manifest/parser.ts

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2023 System Automation Corporation.
+// This file is licensed under the MIT License.
+
+import * as tsdoc from '@microsoft/tsdoc';
+import { Block, parse as parseComment } from 'comment-parser';
+import ts, { SourceFile, SyntaxKind, TypeAliasDeclaration } from 'typescript';
+import debug from './debug';
+
+// Adapted from @microsoft/tsdoc api-demo
+// https://github.com/microsoft/tsdoc/tree/main/api-demo
+
+export interface Comment {
+    compilerNode: ts.Node;
+    commentBlock: Block;
+}
+
+export type ParsedSource = {
+    sourceFile: SourceFile;
+    comments: Comment[];
+    types: {
+        [key: string]: ts.TypeLiteralNode;
+    };
+};
+
+/**
+ * Returns true if the specified SyntaxKind is part of a declaration form.
+ *
+ * Based on ts.isDeclarationKind() from the compiler.
+ * https://github.com/microsoft/TypeScript/blob/v3.0.3/src/compiler/utilities.ts#L6382
+ */
+function isDeclarationKind(kind: ts.SyntaxKind): boolean {
+    return (
```

Comment:
@actong curious. what is this for?

Reply:
This comes from the tsdoc sample https://github.com/microsoft/tsdoc/tree/main/api-demo to only look for comments in declarations.